src/main/java/org/assertj/core/internal/Maps.java

```diff
@@ -796,9 +798,10 @@ public void assertHasSameSizeAs(AssertionInfo info, Map<?, ?> map, Map<?, ?> oth
     if (notExpected.isEmpty() && notFound.isEmpty()) {
       // check entries order
       int index = 0;
-      for (K keyFromActual : actual.keySet()) {
-        if (!areEqual(keyFromActual, entries[index].getKey())) {
-          Map.Entry<K, V> actualEntry = entry(keyFromActual, actual.get(keyFromActual));
+      for (Map.Entry<K, V> entryFromActual : actual.entrySet()) {
+        V value = actual.get(entries[index].getKey());
+        if (!areEqual(value, entryFromActual.getValue())) {
```

Comment:
comparing only the values is not enough, the following test should fail but succeeds with this implementation because all the keys have the same value:
```java
Map<String, String> actual = new LinkedHashMap<>(2);
actual.put("a", "1");
actual.put("b", "1");
maps.assertContainsExactly(INFO, actual, entry("b", "1"), entry("a", "1"));
```
We could solve the problem if we were able to use the actual map way to compare keys to compare `actualKeys[i]` to `entries[i].key` but I'm not sure this is possible.

To be pragmatic I have created https://github.com/assertj/assertj-core/issues/2165 and I will remove the `containsExactly` changes from this PR when integrating (don't bother pushing a commit for that)

Reply:
You are right, I didn't think about that use case. Thanks for removing the changes in the contains exactly method.

Reply:
btw, it would be good to add the test case you shared in this commit to the codebase :)

Reply:
good idea, I'll add it to the other tests